### PR TITLE
Input encoding section

### PR DESCRIPTION
**Pull request by armansito**
Rewrote and rearranged parts of the input encoding section.  Two figures are WIP.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+paper:
+	pdflatex paper & bibtex paper
```
Comment (raphlinus) [inline Makefile]: Should this be `&&`? Also I usually have two more pdflatex invocations, to re-typeset when the bib changes.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+clean:
+	rm paper.aux & rm paper.log & rm paper.out & rm paper.bbl & rm paper.pdf
```
Comment (raphlinus) [inline Makefile]: I think the best way to write this is rm -f <file...>

File: paper.tex [hunk 3]
```diff
@@ -367,43 +381,44 @@ \subsection{Pipeline Design}
 
 \subsection{Input encoding}
 
-% TODO: not sure how much to emphasize the path tag encoding. The input processing monoids benefit the vello scene API (fragments) the TagMonoid could generally be computed on the CPU (with minimal offset tracking) It's not that relevant to the stroke processing. The main bit to mention is that it's a struct-of-arrays.
-% TODO: talk about sanitization of input: closepaths are handled ahead of time, stroke cap marker.
-% TODO: discuss style encoding in greater detail
+The input to our pipeline consists of a sequence of paths, each with an associated 2D affine transformation matrix and a \emph{style} data structure containing information about the stroke style or fill rule. We encode paths, transforms, and styles as parallel input streams. Unlike a conventional structural-of-arrays arrangement, we permit a one-to-many mapping across elements between certain streams: each transform and style entry can apply to one or more path entries with an array index greater than or equal to transform or style index.
```
Comment (raphlinus) [inline paper.tex]: I think this is "structure of arrays"

File: paper.tex [hunk 4]
```diff
@@ -367,43 +381,44 @@ \subsection{Pipeline Design}
...
+The handling of the subpath bit allows for overlap in coordinates, so that except at subpath boundaries, the first coordinate pair of each segment overlaps with the last coordinate pair of the previous one. During encoding, subpath boundaries are moved from the beginning of a subpath (moveto) to the end. An additional line segment is inserted into the tag stream for fills (closepath) if the final point does not coincide with the start point. The final segment of every subpath has the \emph{subpath end bit} set to $1$. For strokes, the subpath end segment represents the \emph{stroke cap marker} defined in Section \ref{subsec:pipeline-design}. This special segment is assigned a coordinate count of $2$, and the corresponding two points in the path data stream encode the tangent vector of the subpath's initial segment, used to correctly draw the start cap or the connecting join in a closed contour.
```
Comment (raphlinus) [inline paper.tex]: I think "is inserted into" is a bit too passive. Maybe "The CPU-side encoding process inserts an additional line segment into the tag and coordinate streams to close the subpath if the final point does not coincide with the start point."

**Review by raphlinus — APPROVED** [review]
I have a few suggestions, but overall happy to see this go in. Onward!

File: paper.tex (hunk 4 above)
Comment (armansito) [reply paper.tex]: I like it. Done.

**Pull request merged**